service/src/main/java/bio/terra/workspace/service/spendprofile/SpendProfile.java

```diff
@@ -18,6 +19,12 @@ public abstract class SpendProfile {
   /** The id of the Google Billing Account associated with the SpendProfile, if there is one. */
   public abstract Optional<String> billingAccountId();
 
+  public abstract Optional<UUID> tenantId();
```

Comment:
Should we have something in this object that says what cloud it is for? That would at least allow verification in other code. Alternately, perhaps we should have a base class and per-cloud subclasses?

If we refactor, I would love to get rid of AutoValue and use a record instead.